host/bcm283x/gpio.go

```diff
@@ -179,6 +180,9 @@ func (p *Pin) In(pull gpio.Pull, edge gpio.Edge) error {
 	if gpioMemory == nil {
 		return p.wrap(errors.New("subsystem not initialized"))
 	}
+	if err := p.haltClock(); err != nil {
```

Comment:
This is a bit tricky when we enable a PWM channel for multiple pins, for example,  12 and 18, then disable it for one of them.

I guess current implementation will disable PWM for both pins when we disable PWM for one of them. This may be different from user's expectation.

I think the current behavior is ok as long as commented somewhere.

Reply:
Thanks for bringing this up. Added comment in Halt(). I'll eventually sort it out but this can wait to a follow up.

Reply:
I agree, that wouldn't have been my expectation either. Great catch!
@maruel I don't think we have a PWM example, maybe you should consider adding your https://gist.github.com/maruel/f0230fce67aa713154d84c9e2e175e1c example to the repo or website?

Reply:
I plan to write an example once it is bug-fixed enough.